app/models/proxy.rb

```diff
@@ -118,6 +118,10 @@ def self.user_attribute_names
     super + %w[api_backend] + GatewayConfiguration::ATTRIBUTES
   end
 
+  def metric_in_latest_configs?(metric_id)
+    ProxyConfig::ENVIRONMENTS.map { |env| proxy_configs.by_environment(env).newest_first.first.contains_metric?(metric_id) }.any?
```

Comment:
I got an `undefined method 'contains_metric?' for nil:NilClass` here. That's because I used a fresh DB and I had no configurations promoted yet. It keeps failing until you have at least one staging and one production configs promoted. Even when it's a corner case, I suggest this change:

```suggestion
    ProxyConfig::ENVIRONMENTS.map { |env| proxy_configs.by_environment(env).newest_first.first&.contains_metric?(metric_id) }.any?
```

Reply:
Good catch! Thanks, I'll fix this